### PR TITLE
Fix missing spec

### PR DESCRIPTION
**Pull request by jakubno**
The openapi.yml was missing in the package, so the validator threw error on every request:
image

https://discord.com/channels/1092243196446249134/1132718082641645608/1136320379049226381

### CODE REVIEW

File: tsconfig.json [hunk 1]
```diff
@@ -8,7 +8,8 @@
     "sourceMap": false,
     "outDir": "./dist",
     "baseUrl": "./",
-    "strictNullChecks": true
+    "strictNullChecks": true,
+    "esModuleInterop": true
```
Comment (jakubno) [inline tsconfig.json:12]: otherwise it didn't work locally

**Comment by ValentaTomas** [comment]
I think this should fix it. I didn't test it thoroughly - do you have a testing "workflow" that you are using locally?

**Comment by ValentaTomas** [comment]
Wait, my bad, there is still an error 🚧

**Comment by jakubno** [comment]
I have separate project, where I have minimal example. I copy the builded package manually into node modules. Now it can't find the spec. I will take look into it. This looks much cleaner, thanks 

```
Error: openapi.validator: spec could not be read at openapi: 3.0.1
info:
  title: Agent Communication Protocol
  description: Specification of the API protocol for communication with an agent.
  version: 'v1'

paths:
  /agent/tasks:
    post:
      summary: Creates a task for the agent.
      tags: [agent]
      operationId: createAgentTask
      requestBody:
        content:
          application/json:
            schema:
              $ref: '#/components/schemas/TaskRequestBody'
      responses:
        200:
          description: A new agent task was successfuly created.
          content:
            application/json:
              schema:
                $ref: '#/components/schemas/Task'
        default:
          description: Internal Server Error
    get:
      summary: List all tasks that have been created for the agent.
      tags: [agent]
      operationId: listAgentTasksIDs
      responses:
        200:
          description: Returned list of agent's task IDs.
          content:
            application/json:
              schema:
                type: array
                default: []
                description: A list of task IDs
                items:
                  type: string
        default:
          description: Internal Server Error

  /agent/tasks/{task_id}:
    get:
      summary: Get details about a specified agent task.
      tags: [agent]
      operationId: getAgentTask
      parameters:
        - name: task_id
          in: path
          description: ID of the task
          required: true
          schema:
            type: string
      responses:
        200:
          description: Returned details about an agent task.
          content:
            application/json:
              schema:
                $ref: '#/components/schemas/Task'
        default:
          description: Internal Server Error

  /agent/tasks/{task_id}/steps:
    get:
      summary: List all steps for the specified task.
      tags: [agent]
      operationId: listAgentTaskSteps
      parameters:
        - name: task_id
          in: path
          description: ID of the task
          required: true
          schema:
            type: string
      responses:
        200:
          description: Returned list of agent's step IDs for the specified task.
          content:
            application/json:
              schema:
                type: array
                default: []
                description: A list of step IDs for the task
                items:
                  type: string
        default:
          description: Internal Server Error
    post:
      summary: Execute a step in the specified agent task.
      tags: [agent]
      operationId: executeAgentTaskStep
      parameters:
        - name: task_id
          in: path
          description: ID of the task
          required: true
          schema:
            type: string
      requestBody:
        content:
          application/json:
            schema:
              $ref: '#/components/schemas/StepRequestBody'
      responses:
        200:
          description: Executed step for the agent task.
          content:
            application/json:
              schema:
                $ref: '#/components/schemas/Step'
        default:
          description: Internal Server Error

  /agent/tasks/{task_id}/steps/{step_id}:
    get:
      summary: Get details about a specified task step.
      tags: [agent]
      operationId: getAgentTaskStep
      parameters:
        - name: task_id
          in: path
          description: ID of the task
          required: true
          schema:
            type: string
        - name: step_id
          in: path
          description: ID of the step
          required: true
          schema:
            type: string
      responses:
        200:
          description: Returned details about an agent task step.
          content:
            application/json:
              schema:
                $ref: '#/components/schemas/Step'
        default:
          description: Internal Server Error

components:
  schemas:
    TaskInput:
      description: Input parameters for the task. Any value is allowed.

    Artifact:
      description: Artifact that the task has produced. Any value is allowed.

    StepInput:
      description: Input parameters for the task step. Any value is allowed.

    StepOutput:
      description: Output that the task step has produced. Any value is allowed.

    TaskRequestBody:
      type: object
      description: Body of the task request.
      properties:
        input:
          $ref: '#/components/schemas/TaskInput'

    Task:
      allOf:
        - $ref: '#/components/schemas/TaskRequestBody'
        - type: object
          description: Definition of an agent task.
          required:
            - 'task_id'
          properties:
            task_id:
              description: The ID of the task.
              type: string
            artifacts:
              type: array
              default: []
              description: A list of artifacts that the task has produced.
              items:
                $ref: '#/components/schemas/Artifact'

    StepRequestBody:
      type: object
      description: Body of the task request.
      properties:
        input:
          $ref: '#/components/schemas/StepInput'

    StepResult:
      description: Result of the task step.
      type: object
      required:
        - 'artifacts'
      properties:
        output:
          $ref: '#/components/schemas/StepOutput'
        artifacts:
          type: array
          default: []
          description: A list of artifacts that the step has produced.
          items:
            $ref: '#/components/schemas/Artifact'
        is_last:
          description: Whether this is the last step in the task.
          type: boolean
          default: false

    Step:
      allOf:
        - $ref: '#/components/schemas/StepRequestBody'
        - $ref: '#/components/schemas/StepResult'
        - type: object
          required:
            - 'step_id'
            - 'task_id'
          properties:
            task_id:
              description: The ID of the task this step belongs to.
              type: string
            step_id:
              description: The ID of the task step.
              type: string

    at OpenAPIFramework.loadSpec (/Users/i/Work/BetterStack/untitled1/node_modules/express-openapi-validator/dist/framework/index.js:66:23)
    at OpenAPIFramework.initialize (/Users/i/Work/BetterStack/untitled1/node_modules/express-openapi-validator/dist/framework/index.js:16:35)
    at OpenApiSpecLoader.discoverRoutes (/Users/i/Work/BetterStack/untitled1/node_modules/express-openapi-validator/dist/framework/openapi.spec.loader.js:27:60)
    at OpenApiSpecLoader.load (/Users/i/Work/BetterStack/untitled1/node_modules/express-openapi-validator/dist/framework/openapi.spec.loader.js:19:21)
    at Object.openapiValidator (/Users/i/Work/BetterStack/untitled1/node_modules/express-openapi-validator/dist/index.js:31:8)
    at Object.<anonymous> (/Users/i/Work/BetterStack/untitled1/node_modules/agent-protocol/dist/index.js:226:177)
    at Module._compile (node:internal/modules/cjs/loader:1198:14)
    at Object.Module._extensions..js (node:internal/modules/cjs/loader:1252:10)
    at Module.load (node:internal/modules/cjs/loader:1076:32)
    at Function.Module._load (node:internal/modules/cjs/loader:911:12)
 ```

**Comment by ValentaTomas** [comment]
I just checked, I have the reason why, just need to fix it.

**Comment by ValentaTomas** [comment]
You need to pass the spec either as a path or parsed object (https://github.com/cdimascio/express-openapi-validator/wiki/Documentation#%EF%B8%8F-apispec-required).
I added the parsing, but there is still some error even though the spec looks conforming to the needed format. Pushing so you can see the latest changes.

```sh
Not Found: not found
    at /Users/valenta.and.thomas/Developer/sdk/agent/js/node_modules/express-openapi-validator/dist/middlewares/openapi.metadata.js:48:19
    at /Users/valenta.and.thomas/Developer/sdk/agent/js/node_modules/express-openapi-validator/dist/openapi.validator.js:125:24
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
```

I also fixed the statically served spec on `/openapi.yaml`.

**Pull request merged**